TestFoundation/TestCodable.swift

```diff
@@ -51,15 +51,17 @@ func expectRoundTripEquality<T : Codable>(of value: T, encode: (T) throws -> Dat
     XCTAssertEqual(value, decoded, "Decoded \(T.self) <\(decoded)> not equal to original <\(value)>")
 }
 
-func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T) where T : Equatable {
+func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T,
```

Comment:
This diverges the code in `swift-corelibs-foundation` from https://github.com/apple/swift/blob/b8dc826d41a25d856d6e827eb67cbfef9c4f9705/test/stdlib/CodableTests.swift#L60

Can we keep things as closely in sync as possible?